File: app/src/main/java/calendario/kevshupp/diariokevinali/ProfileFragment.java
package calendario.kevshupp.diariokevinali;

import android.content.Context;
import android.graphics.Color;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.bumptech.glide.Glide;
import com.google.firebase.firestore.FirebaseFirestore;

import java.util.Calendar;
import java.util.Locale;

public class ProfileFragment extends Fragment {

    private String currentUserId, currentUserName, currentUserImageUri, theme, coupleId;
    private FirebaseFirestore db;

    public static ProfileFragment newInstance(String userId, String userName, String userImageUri, String theme) {
        ProfileFragment fragment = new ProfileFragment();
        Bundle args = new Bundle();
        args.putString("userId", userId);
        args.putString("userName", userName);
        args.putString("userImageUri", userImageUri);
        args.putString("theme", theme);
        fragment.setArguments(args);
        return fragment;
    }

    public static ProfileFragment newInstance(String userId, String userName, String userImageUri, String theme, String coupleId) {
        ProfileFragment fragment = new ProfileFragment();
        Bundle args = new Bundle();
        args.putString("userId", userId);
        args.putString("userName", userName);
        args.putString("userImageUri", userImageUri);
        args.putString("theme", theme);
        args.putString("coupleId", coupleId);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            currentUserId = getArguments().getString("userId");
            currentUserName = getArguments().getString("userName");
            currentUserImageUri = getArguments().getString("userImageUri");
            theme = getArguments().getString("theme");
            coupleId = getArguments().getString("coupleId");
        }
        db = FirebaseFirestore.getInstance();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.dialog_profile, container, false);

        if ("Pixel Oscuro".equals(theme)) {
            v.setBackgroundResource(R.drawable.bg_parchment_pixel_dark);
            ((TextView) v.findViewById(R.id.tvProfileTitle)).setTextColor(Color.WHITE);
            ((TextView) v.findViewById(R.id.tvCurrentUserName)).setTextColor(Color.WHITE);
            TextView tvTogether = v.findViewById(R.id.tvTogetherTime);
            tvTogether.setTextColor(Color.parseColor("#FF80AB"));
            tvTogether.setBackgroundResource(R.drawable.bg_message_pixel_dark);
            ((TextView) v.findViewById(R.id.tvProfileInfo)).setTextColor(Color.LTGRAY);
            
            v.findViewById(R.id.ivProfileImage).setBackgroundResource(R.drawable.bg_parchment_pixel_dark);
            v.findViewById(R.id.ivPartnerImage).setBackgroundResource(R.drawable.bg_parchment_pixel_dark);
            ((TextView) v.findViewById(R.id.tvPartnerName)).setTextColor(Color.parseColor("#FF80AB"));
            v.findViewById(R.id.dividerPartner).setBackgroundColor(Color.parseColor("#444466"));
            
            Button btnSave = v.findViewById(R.id.btnSaveProfile);
            btnSave.setTextColor(Color.WHITE);
            btnSave.setBackgroundTintList(android.content.res.ColorStateList.valueOf(Color.parseColor("#1A1A2E")));
            
            Button btnLogout = v.findViewById(R.id.btnLogoutProfile);
            btnLogout.setTextColor(Color.WHITE);
            btnLogout.setBackgroundTintList(android.content.res.ColorStateList.valueOf(Color.parseColor("#D81B60")));
        } else {
            v.setBackgroundResource(R.drawable.bg_parchment_pixel);
        }

        ImageView ivProfile = v.findViewById(R.id.ivProfileImage);
        ImageView ivPartner = v.findViewById(R.id.ivPartnerImage);
        TextView tvName = v.findViewById(R.id.tvCurrentUserName), tvTime = v.findViewById(R.id.tvTogetherTime);
        TextView tvPartnerName = v.findViewById(R.id.tvPartnerName);
        
        tvName.setText(currentUserName);
        tvTime.setText(calcRelationshipTime(2022, 1, 19));
        
        if (currentUserImageUri != null) {
            Glide.with(this).load(currentUserImageUri).circleCrop().into(ivProfile);
        }

        // Cargar datos de la pareja
        db.collection("users").whereEqualTo("coupleId", coupleId).addSnapshotListener((snapshot, error) -> {
            if (snapshot != null) {
                for (com.google.firebase.firestore.DocumentSnapshot doc : snapshot.getDocuments()) {
                    String userId = doc.getId();
                    // Buscamos al usuario que NO sea el actual para mostrarlo como pareja
                    if (!userId.equals(currentUserId)) {
                        String pName = doc.getString("userName");
                        String pUrl = doc.getString("profileImageUrl");
                        
                        // Si es Ali, nos aseguramos que se vea bien el nombre
                        if (pName != null) {
                            tvPartnerName.setText(pName);
                        } else if (userId.equals("user_ali_02")) {
                            tvPartnerName.setText("Ali");
                        }
                        
                        if (pUrl != null && isAdded()) {
                            Glide.with(this).load(pUrl).circleCrop().into(ivPartner);
                        }
                        break;
                    }
                }
            }
        });

        ivProfile.setOnClickListener(v1 -> {
            if (getActivity() instanceof MainActivity) ((MainActivity) getActivity()).pickImage(1); // PICK_IMAGE_PROFILE
        });

        v.findViewById(R.id.btnSaveProfile).setOnClickListener(v1 -> {
            db.collection("users").document(currentUserId).update("profileImageUrl", currentUserImageUri)
                .addOnSuccessListener(aVoid -> {
                    if (isAdded()) {
                        requireActivity().getSharedPreferences("DiarioPrefs", Context.MODE_PRIVATE).edit().putString("userImage", currentUserImageUri).apply();
                        android.widget.Toast.makeText(getContext(), "Perfil actualizado", android.widget.Toast.LENGTH_SHORT).show();
                    }
                });
        });

        v.findViewById(R.id.btnLogoutProfile).setOnClickListener(v1 -> {
            if (getActivity() instanceof MainActivity) ((MainActivity) getActivity()).logout();
        });

        return v;
    }

    private String calcRelationshipTime(int y, int m, int d) {
        Calendar start = Calendar.getInstance(); start.set(y, m - 1, d); Calendar now = Calendar.getInstance();
        int years = now.get(Calendar.YEAR) - start.get(Calendar.YEAR), months = now.get(Calendar.MONTH) - start.get(Calendar.MONTH), days = now.get(Calendar.DAY_OF_MONTH) - start.get(Calendar.DAY_OF_MONTH);
        if (days < 0) { months--; days += now.getActualMaximum(Calendar.DAY_OF_MONTH); }
        if (months < 0) { years--; months += 12; }
        return String.format(Locale.getDefault(), "Juntos: %d año(s), %d mes(es), %d día(s)", years, months, days);
    }

    public void setProfileImage(String url) {
        this.currentUserImageUri = url;
        if (getView() != null) {
            ImageView ivProfile = getView().findViewById(R.id.ivProfileImage);
            Glide.with(this).load(url).circleCrop().into(ivProfile);
        }
    }
}

File: app/src/main/java/calendario/kevshupp/diariokevinali/ProfileFragment.kt
package calendario.kevshupp.diariokevinali

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue
import androidx.compose.ui.platform.ComposeView
import androidx.fragment.app.Fragment
import calendario.kevshupp.diariokevinali.compose.ProfileScreen
import com.google.firebase.firestore.FirebaseFirestore

class ProfileFragment : Fragment() {
    private var userId: String? = null
    private var partnerId: String? = null
    private var theme: String = "Pixel Claro"
    private var currentUserName: String by mutableStateOf("Kevin")
    private var currentUserImageUri: String? by mutableStateOf(null)

    companion object {
        @JvmStatic
        fun newInstance(userId: String, partnerId: String, theme: String): ProfileFragment {
            val f = ProfileFragment()
            val a = Bundle()
            a.putString("userId", userId)
            a.putString("partnerId", partnerId)
            a.putString("theme", theme)
            f.arguments = a
            return f
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        userId = arguments?.getString("userId")
        partnerId = arguments?.getString("partnerId")
        theme = arguments?.getString("theme") ?: "Pixel Claro"
        loadPrefs()
    }

    fun setProfileImage(url: String) {
        currentUserImageUri = url
    }

    override fun onResume() {
        super.onResume()
        loadPrefs()
    }

    private fun loadPrefs() {
        val prefs = activity?.getSharedPreferences("DiarioPrefs", android.content.Context.MODE_PRIVATE)
        currentUserName = prefs?.getString("userName", currentUserName) ?: currentUserName
        currentUserImageUri = prefs?.getString("userImage", currentUserImageUri)
        theme = prefs?.getString("theme", theme) ?: theme
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        return ComposeView(requireContext()).apply {
            setContent {
                val act = activity as? MainActivity
                ProfileScreen(
                    currentUserName = currentUserName,
                    currentUserImageUri = currentUserImageUri,
                    theme = theme,
                    coupleId = partnerId,
                    currentUserId = userId ?: "",
                    onPickImage = { act?.pickImage(1) },
                    onSaveProfile = { newName, newImage ->
                        if (userId != null) {
                            val db = FirebaseFirestore.getInstance()
                            val updates = mutableMapOf<String, Any>()
                            updates["userName"] = newName
                            updates["userId"] = userId!!
                            if (partnerId != null) updates["coupleId"] = partnerId!!
                            if (newImage != null) updates["profileImageUrl"] = newImage
                            
                            db.collection("users").document(userId!!).set(updates, com.google.firebase.firestore.SetOptions.merge())
                                .addOnSuccessListener {
                                    val prefs = act?.getSharedPreferences("DiarioPrefs", android.content.Context.MODE_PRIVATE)
                                    prefs?.edit()?.putString("userName", newName)?.apply()
                                    if (newImage != null) prefs?.edit()?.putString("userImage", newImage)?.apply()
                                    
                                    currentUserName = newName
                                    act?.runOnUiThread {
                                        Toast.makeText(requireContext(), "Perfil actualizado", Toast.LENGTH_SHORT).show()
                                    }
                                }
                        }
                    },
                    onLogout = { act?.logout() }
                )
            }
        }
    }
}
